src/plugins/intel_cpu/src/emitters/tpp/aarch64/kernel_executors/brgemm.cpp

```diff
@@ -0,0 +1,138 @@
+    return StaticBaseParams::operator==(rhs) && isa == rhs.isa && m_type_in0 == rhs.m_type_in0 &&
+           m_type_in1 == rhs.m_type_in1 && m_type_exec == rhs.m_type_exec && m_type_out0 == rhs.m_type_out0 &&
+           m_compile_flags == rhs.m_compile_flags && m_prefetching_flags == rhs.m_prefetching_flags;
+}
+
+BrgemmKernelExecutor::BrgemmKernelExecutor(ov::intel_cpu::MultiCacheWeakPtr kernel_cache, BrgemmKernelConfig config)
+    : CPUKernelExecutor<BrgemmKernelConfig, BrgemmTppCompiledKernel>(std::move(kernel_cache), std::move(config)) {}
+
+std::shared_ptr<BrgemmTppCompiledKernel> BrgemmKernelExecutor::compile_kernel(const BrgemmKernelConfig& config) const {
+    std::shared_ptr<BrgemmTppCompiledKernel> compiled_kernel = std::make_shared<BrgemmTppCompiledKernel>();
+
+    // Brgemm is not executable - nothing to compile
+    if (config.is_empty())
+        return compiled_kernel;
+
+    libxsmm_gemm_shape m_shape = libxsmm_create_gemm_shape(config.get_N(),
+                                                           config.get_M(),
+                                                           config.get_K(),
+                                                           config.get_LDB(),
+                                                           config.get_LDA(),
+                                                           config.get_LDC(),
+                                                           config.get_type_in0(),
+                                                           config.get_type_in1(),
+                                                           config.get_type_out0(),
+                                                           config.get_type_exec());
+    const auto& compile_flag = config.get_compile_flags();
+    auto refreshed_compile_flag =
+        config.get_beta() == 0 ? config.get_compile_flags() | LIBXSMM_GEMM_FLAG_BETA_0 : compile_flag;
+    compiled_kernel->brgemm_kernel = std::make_shared<libxsmm_gemmfunction>(COMPILE_BRGEMM_TPP_KERNEL(
+        libxsmm_dispatch_gemm(m_shape, refreshed_compile_flag, config.get_prefetching_flags())));
+
+    return compiled_kernel;
+}
+
+void BrgemmKernelExecutor::update_config(const ov::snippets::lowered::ExpressionPtr& expr,
+                                         const ov::snippets::lowered::LinearIRCPtr& linear_ir,
+                                         BrgemmKernelConfig& config) const {
+    BrgemmBaseKernelExecutor::update_config(expr, linear_ir, config);
+    const auto& tpp_mod = std::dynamic_pointer_cast<tpp::modifier::TensorProcessingPrimitive>(expr->get_node());
+    auto replace_full_dim = [](size_t dim, size_t replace_dim) {
+        if (ov::snippets::utils::is_full_dim_value(dim))
+            return replace_dim;
+        return dim;
+    };
+
+    const auto num_ins = expr->get_node()->get_input_size();
+    const auto num_outs = expr->get_node()->get_output_size();
+
+    size_t io_strides[num_ins + num_outs];
```

Comment:
Is it variable length array? If it is, as far as I know, ISO C++ forbids it. Can we use `std::vector` at least here or just 3 size_t variables?

Reply:
`std::vector `is used, Thanks Alexandra!